src/main/java/org/broadinstitute/hellbender/tools/funcotator/mafOutput/MafOutputRenderer.java

```diff
@@ -0,0 +1,764 @@
+     */
+    private LinkedHashMap<String, String> replaceFuncotationValuesWithMafCompliantValues(final Map<String, Object> outputMap ) {
+
+        final LinkedHashMap<String, String> finalOutMap = new LinkedHashMap<>(outputMap.size());
+
+        // Massage individual Key/Value pairs:
+        for ( final String key : outputMap.keySet() ) {
+            finalOutMap.put(key, mafTransform(key, outputMap.get(key).toString()) );
+        }
+
+        // Massage the OtherTranscripts field:
+        finalOutMap.put("Other_Transcripts", finalOutMap.get("Other_Transcripts").replaceAll(VcfOutputRenderer.OTHER_TRANSCRIPT_DELIMITER, "|"));
+
+        // Massage the start/end/alleles in the case of INDELs
+        // (Because MAF has different conventions from VCF for start/end positions of INDELs)
+        if ( finalOutMap.get("Variant_Type").equals("INS") || finalOutMap.get("Variant_Type").equals("DEL") ) {
+
+            final int refAlleleLength = finalOutMap.get("Reference_Allele").length();
+            final int altAlleleLength = finalOutMap.get("Tumor_Seq_Allele2").length();
+
+            // TODO: port these changes to GencodeFuncotationFactory (issue: https://github.com/broadinstitute/gatk/issues/4378)
+            // Check to see if it's an insertion:
+            if ( refAlleleLength < altAlleleLength ) {
+                // We must:
+                //    Remove the first N bases from the ALT_allele where N = length(ref_allele)
+                //    Replace the ref_allele with "-"
+                //    Replace the Tumor_Seq_Allele1 with "-"
+                //    Set the End_Position to be Start_Position + 1 (All Insertions should have length 1 to represent the bases between which the insertion occurs).
+                finalOutMap.put("Tumor_Seq_Allele2", finalOutMap.get("Tumor_Seq_Allele2").substring(refAlleleLength));
+                finalOutMap.put("Reference_Allele", "-");
+                finalOutMap.put("Tumor_Seq_Allele1", "-");
+                finalOutMap.put("End_Position", String.valueOf(Integer.valueOf(finalOutMap.get("Start_Position")) + 1));
+            }
+            // Check to see if it's a deletion:
+            else if ( refAlleleLength > altAlleleLength ) {
+                // We must:
+                //    Remove the first N bases from the REF_allele where N = length(alt_allele)
+                //    Remove the first N bases from the Tumor_Seq_Allele1
+                //    Replace the alt_allele with "-"
+                //    Increment the Start_Position by 1 (start position should be inclusive of the first base deleted)
+                //    Increment the End_Position by M-1 where M = length(ref_allele) (end position should be inclusive of the last base deleted)
+                finalOutMap.put("Reference_Allele", finalOutMap.get("Reference_Allele").substring(altAlleleLength));
+                finalOutMap.put("Tumor_Seq_Allele1", finalOutMap.get("Tumor_Seq_Allele1").substring(altAlleleLength));
```

Comment:
Use constants instead of raw strings, especially for repeated fields.

Reply:
Fixed.